app/controllers/api/solutions/original_solution_files_controller.rb

```diff
@@ -0,0 +1,13 @@
+module API
+  module Solutions
+    class OriginalSolutionFilesController < BaseController
+      def index
+        solution = current_user.solutions.find_by!(uuid: params[:solution_id])
+
+        files = SerializeFiles.(solution.original_solution_files)
+
+        render json: { files: files }
```

Comment:
Please add a controller test that covers the various cases in this. See the other API controller tests for bits you can probably copy/paste in.